File: change_range.py
def change_range(value, old_min, old_max, new_min, new_max):
    old_range = old_max - old_min
    new_range = new_max - new_min
    return float(new_range * (value - old_min)/old_range + new_min)

if __name__ == '__main__':
    old_min = -5
    old_max = 0
    new_min = 0
    new_max = 5
    for v in range(old_min, old_max):
        print change_range(v, old_min, old_max, new_min, new_max)

File: change_range.pl
#!/usr/bin/env perl
use strict;
use warnings;
use feature qw(say);

#simple one-liner
sub change_range_minimal {
    return ($_[4]-$_[3])*($_[0]-$_[1])/($_[2]-$_[1])+$_[3];
}

#in case range is needed
sub change_range {
    my ($value, $oldMin, $oldMax, $newMin, $newMax) = @_;
    my $oldRange = $oldMax - $oldMin;
    my $newRange = $newMax - $newMin;
    return $newRange * ($value - $oldMin)/$oldRange + $newMin;
}

my ($oldMin, $oldMax, $newMin, $newMax) = (-5.0, 0.0, 0.0, 5.0);
for(my $v = $oldMin; $v < $oldMax; $v++) {
    say change_range($v, $oldMin, $oldMax, $newMin, $newMax);
}
